tests/unit/test_snapshot.py

```diff
@@ -223,10 +297,115 @@ def test_iteration_w_raw_raising_unavailable_after_token(self):
         after = _MockIterator(*SECOND)
         request = mock.Mock(test="test", spec=["test", "resume_token"])
         restart = mock.Mock(spec=[], side_effect=[before, after])
-        resumable = self._call_fut(restart, request)
+        database = _Database()
+        database.spanner_api = self._make_spanner_api()
+        session = _Session(database)
+        derived = self._makeDerived(session)
+        resumable = self._call_fut(derived, restart, request)
+        self.assertEqual(list(resumable), list(FIRST + SECOND))
+        self.assertEqual(len(restart.mock_calls), 2)
+        self.assertEqual(request.resume_token, RESUME_TOKEN)
+        self.assertNoSpans()
+
+    def test_iteration_w_raw_w_multiuse(self):
+        from google.cloud.spanner_v1 import (
+            ReadRequest,
+        )
+
+        FIRST = (
+            self._make_item(0),
+            self._make_item(1),
+        )
+        before = _MockIterator(*FIRST)
+        request = ReadRequest(transaction=None)
+        restart = mock.Mock(spec=[], return_value=before)
+        database = _Database()
+        database.spanner_api = self._make_spanner_api()
+        session = _Session(database)
+        derived = self._makeDerived(session)
+        derived._multi_use = True
+        resumable = self._call_fut(derived, restart, request)
+        self.assertEqual(list(resumable), list(FIRST))
+        self.assertEqual(len(restart.mock_calls), 1)
+        begin = 0
+        for call in restart.mock_calls:
+            if "begin" in call.__str__():
```

Comment:
[Calls can be treated like tuples](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args). They also have a `kwargs` property in 3.8+ so you could rewrite this as:
```
begin_count = sum([1 for args in restart.call_args_list if "begin" in args.kwargs])
self.assertEqual(begin_count, 1)
```

Reply:
Done. Modified at other places as well.